R/module_teal_data.R

```diff
@@ -108,14 +98,16 @@ srv_validate_reactive_teal_data <- function(id, # nolint: object_length
   checkmate::assert_flag(validate_shiny_silent_error)
 
   moduleServer(id, function(input, output, session) {
-    data_rv <- reactive(tryCatch(data(), error = function(e) e))
+    data_rv <- reactive(tryCatch(data(), error = function(e) e)) # todo: can be removed as data is already try-catched
```

Comment:
`srv_teal_data()` has `tryCatch` in the end. `srv_teal_data()` is used inside `srv_init_data`.

`srv_validate_reactive_teal_data` is used inside `srv_teal_module` and `srv_teal_data`.
`srv_teal_module` uses data that comes from `srv_init_data`. `srv_init_data` uses `srv_teal_data`. It is `srv_init_data` that has `tryCatch` in the end. 

So input to `srv_validate_reactive_teal_data` in `srv_teal_module` is tryCatch-ed but input to `srv_validate_reactive_teal_data` in `srv_teal_data` is not tryCatch-ed

Reply:
```
srv_teal_module <- function() {
  x <- srv_init_data() #input is tryCatch-ed
  srv_validate_reactive_teal_data(x)
}

srv_init_data <- function() {
   y <- srv_teal_data()
   tryCatch(y)
}

srv_teal_data <- function() {
   srv_validate_reactive_teal_data() #input is not tryCatched
}
```

Reply:
Hmm, we can figure out where exactly to put tryCatch to minimise this.

Reply:
It's probably rather like if we have `tryCatch` in `srv_validate_reactive_teal_data` then we don't need a `tryCatch` in `srv_init_data` at the end